### PR TITLE
Refactor to use new cmc-ccd-domain jar

### PR DESCRIPTION
**Pull request by markry-kainos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5877

### Change description ###

Lift n Shift CCD model out of claim-store.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/StreamUtil.java [hunk 1]
```diff
@@ -1,18 +0,0 @@
-package uk.gov.hmcts.cmc.ccd.util;
-
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Optional;
-import java.util.stream.Stream;
-
-public class StreamUtil {
```
Comment (sabahirfan) [inline ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/util/StreamUtil.java]: Please keep this here

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/adapter/util/SampleData.java [hunk 2]
```diff
@@ -472,9 +470,9 @@ public static CCDCase getCCDLegalCase() {
             .amountType(RANGE)
             .amountLowerValue("5000")
             .amountHigherValue("50000")
-            .housingDisrepairCostOfRepairDamages(MORE_THAN_THOUSAND_POUNDS.name())
-            .housingDisrepairOtherDamages(THOUSAND_POUNDS_OR_LESS.name())
-            .personalInjuryGeneralDamages(MORE_THAN_THOUSAND_POUNDS.name())
+            .housingDisrepairCostOfRepairDamages("MORE_THAN_THOUSAND_POUNDS")
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/adapter/util/SampleData.java]: Prefer not to use strings here, would be helpful in refactoring if using enum constants here. You may revert this change.

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/adapter/util/SampleData.java [hunk 3]
```diff
@@ -85,9 +83,9 @@
         .externalId(UUID.randomUUID().toString())
         .features("admissions")
         .amountType(BREAK_DOWN)
-        .housingDisrepairCostOfRepairDamages(MORE_THAN_THOUSAND_POUNDS.name())
-        .housingDisrepairOtherDamages(THOUSAND_POUNDS_OR_LESS.name())
-        .personalInjuryGeneralDamages(MORE_THAN_THOUSAND_POUNDS.name())
+        .housingDisrepairCostOfRepairDamages("MORE_THAN_THOUSAND_POUNDS")
```
Comment (sabahirfan) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/adapter/util/SampleData.java]: Prefer not to use strings here,

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/adapter/util/SampleData.java [hunk 4]
```diff
@@ -146,15 +144,15 @@ public static CCDResponseAcceptation getResponseAcceptationWithClaimantPaymentIn
...
+            .submittedOn(LocalDateTime.now(ZoneId.of("Europe/London")))
             .formaliseOption(SETTLEMENT)
             .build();
     }
 
     public static CCDResponseRejection getResponseRejection() {
         return CCDResponseRejection.builder()
             .amountPaid(AMOUNT)
-            .submittedOn(LocalDateTimeFactory.nowInLocalZone())
+            .submittedOn(LocalDateTime.now(ZoneId.of("Europe/London")))
```
Comment (andrewwa-kainos) [inline ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/adapter/util/SampleData.java]: Will these changes fix ROC-6102?

File: ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/App.java [hunk 5]
```diff
@@ -16,7 +16,7 @@
 @SpringBootApplication(scanBasePackages = {
     "uk.gov.hmcts.reform.authorisation",
     "uk.gov.hmcts.cmc.ccd.migration",
-    "uk.gov.hmcts.cmc.ccd.mapper"
+    "uk.gov.hmcts.cmc.ccd_adapter.mapper"
```
Comment (karthikkannankainos) [inline ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/App.java]: ccd.adapter? Not sure if _ will be accepted, but for consistency pls make it `.`

Comment (andrewwa-kainos) [reply ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/App.java]: Underscore is recommended in cases where a hyphen would normally be used - so depends on if this is meant to be `ccd-adapter` or `ccd adapter` (space).
https://docs.oracle.com/javase/tutorial/java/package/namingpkgs.html

File: ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/adapter/util/SampleData.java (hunk 4 above)
Comment (karthikkannankainos) [reply ccd-adapter/src/test/java/uk/gov/hmcts/cmc/ccd/adapter/util/SampleData.java]: SampleData. This wont fix I think.

File: ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/App.java (hunk 5 above)
Comment (markry-kainos) [reply ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/App.java]: its already changed to `ccd.adapter` - checkstyle pulled it too 😊
